DFT/jesus_dft.py

```diff
@@ -574,3 +575,340 @@ def show_signal_cosines(self):
+        )
+        aux_signal_axis.scale_to_fit_width(cos_mob.width).move_to(cos_mob)
+
+        dots_analysis_freq = get_sampled_dots(
+            analysis_freq_mob, aux_analysis_freq_axis, num_points=10
+        ).set_fill(REDUCIBLE_VIOLET)
+        dots_cos_mob = get_sampled_dots(
+            sine_wave, aux_signal_axis, num_points=10
+        ).set_fill(REDUCIBLE_YELLOW)
+
+        self.play(LaggedStartMap(FadeIn, [*dots_analysis_freq, *dots_cos_mob]))
+
+        dots_line = dots_cos_mob.copy()
+        [
+            d.set_fill(REDUCIBLE_GREEN_LIGHTER, opacity=1).move_to(
+                DOWN * 2.5, coor_mask=[0, 1, 0]
+            )
+            for d in dots_line
+        ]
+
+        self.play(
+            Transform(dots_cos_mob, dots_line), Transform(dots_analysis_freq, dots_line)
+        )
+        self.wait()
+
+        zero = (
+            Text(str(0), font=REDUCIBLE_MONO, weight=SEMIBOLD)
+            .scale(2)
+            .move_to(dots_cos_mob)
+        )
+
+        self.play(Transform(dots_cos_mob, zero), FadeOut(dots_analysis_freq))
+        self.wait()
+
+    def test_cases_again(self):
+        frame = self.camera.frame
+        t_min = 0
+        t_max = TAU * 2
+
+        # samples per second
+        sample_frequency = 80
+
+        # total number of samples
+        n_samples = sample_frequency
+
+        duration = n_samples / sample_frequency
+
+        analysis_frequencies = [
+            sample_frequency * m / n_samples for m in range(n_samples // 2)
+        ]
+
+        # let's just take one AF as an example
+        original_freq = analysis_frequencies[2]
+
+        # this tracker will move phase: from 0 to PI/2
+        vt_frequency = ValueTracker(original_freq)
+        vt_phase = ValueTracker(0)
+        vt_amplitude = ValueTracker(1)
+        vt_b = ValueTracker(0)
+
+        def change_text_redraw():
+            v_freq = f"{vt_frequency.get_value():.2f}"
+            v_amplitude = f"{vt_amplitude.get_value():.2f}"
+            v_phase = f"{degrees(vt_phase.get_value()) % 360:.2f}"
+            v_b = f"{vt_b.get_value():.2f}"
+
+            tex_frequency = Text(
+                "ƒ = " + v_freq + " Hz",
+                font=REDUCIBLE_FONT,
+                t2f={v_freq: REDUCIBLE_MONO},
+            ).scale(0.8)
+
+            phi_eq = MathTex(r"\phi = ")
+            tex_phase_n = Text(
+                v_phase + "º",
+                font=REDUCIBLE_FONT,
+                t2f={v_phase: REDUCIBLE_MONO},
+            ).scale(0.8)
+            tex_phase = VGroup(phi_eq, tex_phase_n).arrange(RIGHT)
+
+            tex_amplitude = Text(
+                "A = " + v_amplitude,
+                font=REDUCIBLE_FONT,
+                t2f={v_amplitude: REDUCIBLE_MONO},
+            ).scale(0.8)
+
+            tex_b = Text(
+                "b = " + v_b,
+                font=REDUCIBLE_FONT,
+                t2f={v_b: REDUCIBLE_MONO},
+            ).scale(0.8)
```

Comment:
I wonder what this would look like if we kept the `b = `, `A = `, etc as a Tex object and then `VGroup` it with a Text object of the actual values in `REDUCIBLE_MONO`. I think it would look possibly better.

Reply:
I think they would look the same since with our Text (not Tex) object we are using the same font as the Tex object does. It simply works orders of magnitude faster because it does not need to render the LaTeX .dvi files. Text objects are so much faster but I made that decision taking into account the "math font" :P